src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -233,21 +246,44 @@ private ReSharperStatus IsReSharperEnabled()
             }
 
             var cmds = new OLECMD[1];
-            cmds[0].cmdID = SuspendId;
+            cmds[0].cmdID = ResumeId;
             cmds[0].cmdf = 0;
 
-            var hr = _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
-            if (ErrorHandler.Failed(hr))
+            for (var count = 0; count < 10; count++)
             {
-                // In the case of an error when attempting to get the status, pretend that ReSharper isn't enabled. We also
-                // shut down monitoring so we don't keep hitting this.
-                FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
-                Shutdown();
-                return ReSharperStatus.NotInstalledOrDisabled;
+                var hr = _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
+                if (ErrorHandler.Failed(hr))
+                {
+                    // In the case of an error when attempting to get the status, pretend that ReSharper isn't enabled. We also
+                    // shut down monitoring so we don't keep hitting this.
+                    FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
+                    Shutdown();
+                    return ReSharperStatus.NotInstalledOrDisabled;
+                }
+
+                // When ReSharper is suspended, the ReSharper_Resume command has the Enabled | Supported flags. 
+                if (((OLECMDF) cmds[0].cmdf).HasFlag(OLECMDF.OLECMDF_ENABLED))
+                {
+                    return ReSharperStatus.Suspended;
+                }
+
+                //otherwise sleep for a bit and check again
```

Comment:
if you move usage of _oleCommandTarget to its own method that switch to UI thread, then, you can keep using Task.Delay with the cancellation token? and since cancellation token is only for shutdown? you might just need 1 cancellation token or 1 each time command state changes?